File: 19/Review/app/src/main/java/com/wx/review/mvp/base/BaseActivity.java
package com.wx.review.mvp.base;


import android.app.Activity;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

public abstract class BaseActivity extends Activity implements IBaseView{

    private boolean isDestroy = false;
    protected final String TAG = this.getClass().getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setDataBinding(DataBindingUtil.setContentView(this,getLayoutId()));
        initData();
        if (getPresenter() != null) {
            getPresenter().onStart();
        }
    }

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }

    protected void openActivity(Activity activity) {

    }

    protected void showToast(String data) {

    }

    protected void showToast(@StringRes int data) {

    }

    @Override
    public boolean isDestroy() {
        return isDestroy;
    }

    @Override
    protected void onDestroy() {
        isDestroy = true;
        super.onDestroy();
        if (getPresenter() != null) {
            getPresenter().onDestroy();
        }
    }

}

File: 19/ACC/AccOne/app/src/main/java/com/wx/acc/base/BaseActivity.kt
package com.wx.acc.base

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

abstract class BaseActivity : AppCompatActivity() {

    protected val TAG = javaClass.simpleName

    @get: LayoutRes
    abstract val layoutId: Int


    abstract fun initData(binding: ViewDataBinding);

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initData(DataBindingUtil.setContentView(this,layoutId))
    }

    fun openActivity(otherActivity: Class<out Activity> ) {
        startActivity(Intent(this,otherActivity))
    }


}